utils/utils_test.go

```diff
@@ -1,10 +1,195 @@
 package utils
 
 import (
+	"context"
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
 	"github.com/stretchr/testify/require"
 	"testing"
+	"time"
 )
 
+func TestBatchSliceOfStrings(t *testing.T) {
+	subtests := []struct {
+		name   string
+		keys   []string
+		count  int
+		output [][]string
+	}{
+		{"nil", nil, 1, nil},
+		{"empty", make([]string, 0, 1), 1, nil},
+		{"a", []string{"a"}, 1, [][]string{{"a"}}},
+		{"a2", []string{"a"}, 2, [][]string{{"a"}}},
+		{"a_b", []string{"a", "b"}, 1, [][]string{{"a"}, {"b"}}},
+		{"ab", []string{"a", "b"}, 2, [][]string{{"a", "b"}}},
+		{"ab3", []string{"a", "b"}, 3, [][]string{{"a", "b"}}},
+		{"a_b_c", []string{"a", "b", "c"}, 1, [][]string{{"a"}, {"b"}, {"c"}}},
+		{"ab_c", []string{"a", "b", "c"}, 2, [][]string{{"a", "b"}, {"c"}}},
+		{"abc", []string{"a", "b", "c"}, 3, [][]string{{"a", "b", "c"}}},
+		{"abc4", []string{"a", "b", "c"}, 4, [][]string{{"a", "b", "c"}}},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			batches := BatchSliceOfStrings(context.Background(), st.keys, st.count)
+			require.NotNil(t, batches)
+
+			for _, expected := range st.output {
+				select {
+				case actual, ok := <-batches:
+					require.True(t, ok, "receiving should return a value")
+					require.Equal(t, expected, actual)
+				case <-time.After(10 * time.Millisecond):
+					require.Fail(t, "receiving should not block")
+				}
+			}
+
+			select {
+			case _, ok := <-batches:
+				require.False(t, ok, "receiving from channel should not return anything")
+			case <-time.After(10 * time.Millisecond):
+				require.Fail(t, "receiving should not block")
+			}
+		})
+	}
+}
+
+func TestChecksum(t *testing.T) {
+	subtests := []struct {
+		name  string
+		input any
+	}{
+		{"empty_string", ""},
+		{"empty_bytes", []byte(nil)},
+		{"space_string", " "},
+		{"space_bytes", []byte(" ")},
+	}
```

Comment:
The `Checksum` function panics for other `input` types. Please test those as well.